code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -419,16 +419,24 @@
 		return src.attack_hand(user)
 	..(user)
 
-/obj/structure/closet/attack_animal(mob/living/simple_animal/user as mob)
-	if(user.environment_smash_flags & SMASH_CONTAINERS)
-//		user.do_attack_animation(src, user) //This will look stupid
+/obj/structure/closet/attack_animal(var/mob/living/user)
+	if(istype(user,/mob/living/simple_animal))
+		var/mob/living/simple_animal/M=user
+		if(M.environment_smash_flags & SMASH_CONTAINERS)
+		//	user.do_attack_animation(src, user) //This will look stupid
+			visible_message("<span class='warning'>[user] destroys the [src]. </span>")
+			broken = 1
+			if(has_electronics)
+				dump_electronics()
+			dump_contents()
+			qdel(src)
+	else if(istype(user,/mob/living/simple_animal))
```

Comment:
istype simple animal followed by else istype simpleanimal, code will not fire. Likely means complex_animal here